src/v/kafka/server/handlers/topics/validators.h

```diff
@@ -282,6 +268,82 @@ struct subject_name_strategy_validator {
     }
 };
 
+struct write_caching_configs_validator {
+    static constexpr const char* error_message
+      = "Unsupported write caching configuration.";
+    static constexpr const auto config_name = topic_property_write_caching;
+    static constexpr error_code ec = error_code::invalid_config;
+
+    static bool validate_write_caching(const creatable_topic& c) {
+        auto it = std::find_if(
+          c.configs.begin(),
+          c.configs.end(),
+          [](const createable_topic_config& cfg) {
+              return cfg.name == topic_property_write_caching;
+          });
+        if (it == c.configs.end() || it->value.has_value()) {
+            return true;
+        }
+        auto mode = model::write_caching_mode_from_string(it->value.value());
+        if (!mode) {
+            // nothing is set.
+            return true;
+        }
+        auto is_user_topic = model::is_user_topic(
+          model::topic_namespace{model::kafka_namespace, c.name});
+        if (is_user_topic) {
+            // disabled mode only allowed globally and cannot be set at topic
+            // level.
+            return mode != model::write_caching_mode::disabled;
+        }
+        // write caching cannot be turned on for internal topics.
+        return mode != model::write_caching_mode::on;
+    }
+
+    static bool validate_flush_ms(const creatable_topic& c) {
+        auto it = std::find_if(
+          c.configs.begin(),
+          c.configs.end(),
+          [](const createable_topic_config& cfg) {
+              return cfg.name == topic_property_flush_ms;
+          });
+        if (it == c.configs.end() || it->value.has_value()) {
+            return true;
+        }
+        try {
+            auto val = boost::lexical_cast<std::chrono::milliseconds::rep>(
+              it->value.value());
+            // atleast 1ms, anything less than that is suspiciously small.
+            return val >= 1;
+        } catch (...) {
+        }
+        return false;
+    }
+
+    static bool validate_flush_bytes(const creatable_topic& c) {
+        auto it = std::find_if(
+          c.configs.begin(),
+          c.configs.end(),
+          [](const createable_topic_config& cfg) {
+              return cfg.name == topic_property_flush_bytes;
+          });
+        if (it == c.configs.end() || it->value.has_value()) {
+            return true;
+        }
+        try {
+            auto val = boost::lexical_cast<size_t>(it->value.value());
+            return val > 0;
```

Comment:
do we always duplicate the validation logic in 2 places like this? also seen this in config_utils.h

Reply:
Thats true, thats how it is done today. The two cases are topic creation and updating properties, the validators use different signatures but we could probably consolidate the logic somehow by factoring out the common validation logic.